app/src/main/java/com/ggaebiz/ggaebiz/presentation/ui/home/HomeScreen.kt

```diff
@@ -1,9 +1,150 @@
 
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.ggaebiz.ggaebiz.R
+import com.ggaebiz.ggaebiz.presentation.designsystem.component.button.GaeBizButton
+import com.ggaebiz.ggaebiz.presentation.designsystem.component.header.GaeBizLogoAppBar
+import com.ggaebiz.ggaebiz.presentation.designsystem.theme.GaeBizTheme
+import com.ggaebiz.ggaebiz.presentation.designsystem.ui.GaeBizMent
+import com.ggaebiz.ggaebiz.presentation.designsystem.ui.GaeBizTag
+import com.ggaebiz.ggaebiz.presentation.feature.Character.Companion.CHARACTER_LIST
 
 @Composable
-fun HomeScreen() {
-    Text("Home")
-}
\ No newline at end of file
+fun HomeScreen(
+    modifier: Modifier = Modifier,
+    navigateSetting: (Int) -> Unit = { },
+) {
+    val pagerState = rememberPagerState(initialPage = 0, pageCount = { CHARACTER_LIST.size })
+    var selectedCharacterIndex by remember { mutableIntStateOf(pagerState.currentPage) }
+    val selectedCharacter = CHARACTER_LIST[pagerState.currentPage]
+
+    val configuration = LocalConfiguration.current
+    val screenWidth = configuration.screenWidthDp.dp
+    val imageWidth = screenWidth / 3 * 2
+    val sideOffset = screenWidth / 6
+
+    LaunchedEffect(pagerState.currentPage) {
+        snapshotFlow { pagerState.currentPage }.collect { page ->
+            selectedCharacterIndex = page
+        }
+    }
```

Comment:
```suggestion
    LaunchedEffect(Unit) {
        snapshotFlow { pagerState.currentPage }
            .collectLatest { page ->
                selectedCharacterIndex = page
            }
    }
```
현재 구조에서는 
 1. snapshotFlow 자체에서도 관찰 
 2. collect{} 람다에서 재수집
 3. pagerState.currentPage 변경될때마다 flow등록  

으로 중복 탐지를 하고 있어서 수정 제안 요청드립니다~

Reply:
오,,,! 좋아 예리해